### ISSUE TITLE
Failed installing Solidity on macOS 10.12.16

### ISSUE THREAD

**Issue by PhDuck** [ethereum/homebrew-ethereum#163]
I have a previous installation 0.4.18 and I'm trying to upgrade to 0.4.21.

macOS Sierra version 10.12.16

Here is the gist-logs https://gist.github.com/PhDuck/4e927000123e6bbbb4f0d46c6ebcceb7#file-solidity-2018-04-05_13-00-41-txt

When I first tried to update it failed because I was missing the newest version of Xcode. After updating xcode the error message is now the following:

```
Last 15 lines from /Users/mg/Library/Logs/Homebrew/solidity/02.make:
Undefined symbols for architecture x86_64:
  "Json::Value::append(Json::Value&&)", referenced from:
      dev::solidity::CompilerStack::createMetadata(dev::solidity::CompilerStack::Contract const&) const in libsolidity.a(CompilerStack.cpp.o)
      dev::solidity::ASTJsonConverter::appendMove(Json::Value&, Json::Value&&) in libsolidity.a(ASTJsonConverter.cpp.o)
      dev::solidity::ASTJsonConverter::visit(dev::solidity::SourceUnit const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      Json::Value dev::solidity::ASTJsonConverter::toJson<dev::solidity::ASTNode>(std::__1::vector<std::__1::shared_ptr<dev::solidity::ASTNode>, std::__1::allocator<std::__1::shared_ptr<dev::solidity::ASTNode> > > const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      dev::solidity::ASTJsonConverter::visit(dev::solidity::PragmaDirective const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      Json::Value dev::solidity::ASTJsonConverter::getContainerIds<std::__1::vector<dev::solidity::ContractDefinition const*, std::__1::allocator<dev::solidity::ContractDefinition const*> > >(std::__1::vector<dev::solidity::ContractDefinition const*, std::__1::allocator<dev::solidity::ContractDefinition const*> > const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      Json::Value dev::solidity::ASTJsonConverter::toJson<dev::solidity::InheritanceSpecifier>(std::__1::vector<std::__1::shared_ptr<dev::solidity::InheritanceSpecifier>, std::__1::allocator<std::__1::shared_ptr<dev::solidity::InheritanceSpecifier> > > const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [solc/solc] Error 1
make[1]: *** [solc/CMakeFiles/solc.dir/all] Error 2
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/ethereum/homebrew-ethereum/issues

Error: undefined method `check_for_bad_install_name_tool' for #<Homebrew::Diagnostic::Checks:0x000001020f3720>
Did you mean?  check_for_tap_ruby_files_locations
Please report this bug:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Homebrew/exceptions.rb:426:in `block in dump'
/usr/local/Homebrew/Library/Homebrew/exceptions.rb:425:in `each'
/usr/local/Homebrew/Library/Homebrew/exceptions.rb:425:in `dump'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:157:in `rescue in upgrade_formula'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:169:in `upgrade_formula'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:95:in `block in upgrade'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:92:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:92:in `upgrade'
/usr/local/Homebrew/Library/Homebrew/brew.rb:100:in `<main>'
```

**Comment by axic** [comment]
@chfast I think this somehow manages to pick up a wrong version of the jsoncpp header files, probably the same issue as https://github.com/ethereum/solidity/pull/3467

Any ideas?

**Comment by chfast** [comment]
Yes. The system include path is before Hunter/jsoncpp include path. In the end you include system headers of jsoncpp.

You have track down who is including `/usr/include`. I remember in cpp-ethereum this path was included unconditionally on macOS.

**Comment by axic** [comment]
@chfast do you want to fix it for https://github.com/ethereum/solidity/pull/3467 ? :)

**Comment by chfast** [comment]
If it's in Homebrew there is not much I can do about. All the headers are in the same dir.
But I'm not sure it's headers issue.

**Comment by axic** [comment]
It is not only Homebrew, it fails on Travis with newer emscripten too. That is the linked issue.

**Comment by kasima** [comment]
I'm having this problem building from source as well. I've checked out the v0.4.21 tag and got this on `make`:

```
[ 63%] Linking CXX executable solc
Undefined symbols for architecture x86_64:
  "Json::Value::append(Json::Value&&)", referenced from:
      dev::solidity::CompilerStack::createMetadata(dev::solidity::CompilerStack::Contract const&) const in libsolidity.a(CompilerStack.cpp.o)
      dev::solidity::ASTJsonConverter::appendMove(Json::Value&, Json::Value&&) in libsolidity.a(ASTJsonConverter.cpp.o)
      dev::solidity::ASTJsonConverter::visit(dev::solidity::SourceUnit const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      Json::Value dev::solidity::ASTJsonConverter::toJson<dev::solidity::ASTNode>(std::__1::vector<std::__1::shared_ptr<dev::solidity::ASTNode>, std::__1::allocator<std::__1::shared_ptr<dev::solidity::ASTNode> > > const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      dev::solidity::ASTJsonConverter::visit(dev::solidity::PragmaDirective const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      Json::Value dev::solidity::ASTJsonConverter::getContainerIds<std::__1::vector<dev::solidity::ContractDefinition const*, std::__1::allocator<dev::solidity::ContractDefinition const*> > >(std::__1::vector<dev::solidity::ContractDefinition const*, std::__1::allocator<dev::solidity::ContractDefinition const*> > const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      Json::Value dev::solidity::ASTJsonConverter::toJson<dev::solidity::InheritanceSpecifier>(std::__1::vector<std::__1::shared_ptr<dev::solidity::InheritanceSpecifier>, std::__1::allocator<std::__1::shared_ptr<dev::solidity::InheritanceSpecifier> > > const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [solc/solc] Error 1
make[1]: *** [solc/CMakeFiles/solc.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by kasima** [comment]
Turns out, I had brew installed jsoncpp. I was able to get past this issue with `brew uninstall jsoncpp`.

**Comment by axic** [comment]
Should be fixed by 0.4.22.

**Comment by axic** [comment]
Will be completely fixed in the next release after 0.4.23.

**Comment by PhDuck** [comment]
Thanks for all the work :)
